multimodal/src/autogluon/multimodal/utils.py

```diff
@@ -2033,3 +2034,41 @@ def process_save_path(path, resume: Optional[bool] = False, raise_if_exist: Opti
             path = None
 
     return path
+
+
+def compute_num_gpus(config_num_gpus: Union[int, float, List], strategy: str):
+    """
+    Compute the gpu number to initialize the lightning trainer.
+
+    Parameters
+    ----------
+    config_num_gpus
+        The gpu number provided by config.
+    strategy
+        A lightning trainer's strategy such as "ddp", "ddp_spawn", and "dp".
+
+    Returns
+    -------
+    A valid gpu number for the current environment and config.
+    """
+    config_num_gpus = (
+        math.floor(config_num_gpus) if isinstance(config_num_gpus, (int, float)) else len(config_num_gpus)
+    )
+    detected_num_gpus = torch.cuda.device_count()
+
+    if config_num_gpus < 0:  # In case config_num_gpus is -1, meaning using all gpus.
+        num_gpus = detected_num_gpus
+    else:
+        num_gpus = min(config_num_gpus, detected_num_gpus)
```

Comment:
Maybe give a warning here saying detected less gpu than what's in the config?

Reply:
Sure. Added a warning.